homework-selects/sql_questions.sql

```diff
@@ -3,55 +3,460 @@
+    INNER JOIN jobs USING (job_id)
+WHERE salary > (
+    SELECT MIN(average_salaries_table.average_salary)
+    FROM (
+         SELECT AVG(salary) AS average_salary
+         FROM EMPLOYEES
+         ) average_salaries_table
+    );
+
 --31. the employee id, name ( first name and last name ) and the job id column with a modified title SALESMAN for those employees whose job title is ST_MAN and DEVELOPER for whose job title is IT_PROG.
+SELECT employee_id, (first_name || ' ' || last_name) AS employee_name,
+       CASE WHEN job_id = 'ST_MAN' THEN 'SALESMAN'
+           WHEN job_id = 'IT_PROG' THEN 'DEVELOPER'
+           ELSE job_id END AS job_id
+FROM employees;
+
 --32. the employee id, name ( first name and last name ), salary and the SalaryStatus column with a title HIGH and LOW respectively for those employees whose salary is more than and less than the average salary of all employees.
+SELECT employee_id, (first_name || ' ' || last_name) AS employee_name,
+       CASE WHEN salary >= (SELECT AVG(salary)
+                            FROM employees) THEN 'HIGH'
+            ELSE 'LOW' END AS SalaryStatus
+FROM employees;
+
 --33. the employee id, name ( first name and last name ), SalaryDrawn, AvgCompare (salary - the average salary of all employees)
     -- and the SalaryStatus column with a title HIGH and LOW respectively for those employees whose salary is more than and less than
     -- the average salary of all employees.
+SELECT employee_id, (first_name || ' ' || last_name) AS employee_name,
+       salary * commission_pct AS SalaryDrawn,
+       CASE WHEN salary >= avg_salary THEN 'HIGH'
+          ELSE 'LOW' END AS SalaryStatus,
+       salary - avg_salary as AvgCompare
+FROM employees,
+    (
+    SELECT AVG(salary) AS avg_salary
+    FROM employees
+);
+
 --34. all the employees who earn more than the average and who work in any of the IT departments.
+SELECT *
+FROM employees
+WHERE salary > (
+    SELECT AVG (salary)
+    FROM employees
+)
+  AND department_id IN (
+    SELECT department_id
+    FROM departments
+    WHERE department_name LIKE '%IT%'
+);
+
 --35. who earns more than Mr. Ozer.
+SELECT *
+FROM employees
+WHERE salary > (
+    SELECT salary
+    FROM employees
+    WHERE last_name = 'Ozer'
+    );
+
 --36. which employees have a manager who works for a department based in the US.
+SELECT *
+FROM employees
+WHERE manager_id IN (
+    SELECT manager_id
+    FROM departments
+    WHERE location_id IN (
+        SELECT location_id
+        FROM locations
+        WHERE country_id  = 'US'
+    ) AND manager_id IS NOT NULL
+);
+
 --37. the names of all employees whose salary is greater than 50% of their department’s total salary bill.
+SELECT (first_name || ' ' || last_name) AS employee_name
+FROM employees
+         INNER JOIN (
+             SELECT department_id, SUM(salary) AS total_salary
+             FROM employees
+             GROUP BY department_id
+) USING (department_id)
+WHERE salary > total_salary * 0.5;
```

Comment:
Rather than using both a JOIN and a SUB QUERY, use only a SUB QUERY.
Your sub-query should extract 1/2 * AVG(SUM) of the department, smth like:
```suggestion
FROM employees emp
WHERE emp.salary > (SELECT SUM(salary) / 2 FROM employees WHERE department_id = emp.department_id)
```

Reply:
Refactored to use only a **SUB QUERY**